### ISSUE TITLE
Nextcloud does not detect that files are accessible from the internet

### ISSUE THREAD

**Issue by arch-user-france1** [nextcloud/server#32955]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

If a .htaccess file is in the data folder the nextcloud overview won't say that the files are accessible from the internet. I could download the nextcloud log but nextcloud said that the files were not accessible from the internet.

### Steps to reproduce

1. 
2.
3.

### Expected behavior

It will show the message.

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
not important
```

### List of activated Apps

```shell
not relevant
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
I prefer to not show these
```

**Comment by kesselb** [comment]
I agree that would be a nice enhancement. We have already a check during setup (that creation of .htaccess was successfull and it does protect the folder in general). But when someone alter or remove the .htaccess there is no other check.

Feel free to dive into the code a bit and sent a pull request ;)

**Comment by joshtrichards** [comment]
I'm not sure what this was referring to. Even in 2022 we had a setup check for the data directory being accessible. It checked for the `.ocdata` file being accessible (not the mere presence of the `.htaccess` file) -- which would be essentially equivalent to trying to retrieve `nextcloud.log` 

Today the setup check is server-side, but largely similar:

https://github.com/nextcloud/server/blob/master/apps/settings/lib/SetupChecks/DataDirectoryProtected.php

Am I missing something?

**Comment by arch-user-france1** [comment]
Seems fine to me
